### DISCUSSION TITLE
undefined method sign for "":String

### DISCUSSION THREAD

**Discussion by a14guguliye** [janko/rodauth-rails#147] [Q&A]
Hello Team,
i have below issue and not sure whether it is related to rodauth-rails or rodauth-oauth 
the issue happens after insert into oauth_tokens is done and this rolls back the transaction 
can someone please help why that happens? may be a hint?

<img width="491" alt="Capture" src="https://user-images.githubusercontent.com/53395754/185174371-52a35b69-6685-434d-854e-9b9be6125c71.PNG">

**Comment by janko** [comment]
Can you post the full backtrace of the exception?

**Reply by a14guguliye** [reply]
unfortunately, thats all I get.

**Reply by janko** [reply]
That's strange, Rails must be truncating the rest of the backtrace, probably because it's not application trace. You'll need to find a way to get the full backtrace. I'd recommend maybe getting the error presented in default Rails exception viewer, there you have tabs that allow you to view the whole backtrace.

I'll also need any relevant Rodauth configuration you have, especially related to rodauth-oauth. It seems so far like an issue with rodauth-oauth, not rodauth-rails, but this additional information should tell us more.

**Reply by a14guguliye** [reply]
do I have any other means to see which part of the package is calling the sign method? the error clearly says that undefined method of sign for empty string

**Reply by janko** [reply]
Yes, search through rodauth-oauth code, and find a call to sign. Unless proven otherwise, this is not an issue in rodauth-rails. If Rails is filtering the exception backtrace, you need to figure out how to get the rest and debug the problem.
